lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -66,6 +68,46 @@ class AwsCompileFunctions {
     }
   }
 
+  downloadPackageArtifact(functionName) {
+    const { region } = this.options;
+    const S3 = new AWS.S3({ region });
+
+    const functionObject = this.serverless.service.getFunction(functionName);
+    const artifactFilePath = _.get(functionObject, 'package.artifact') ||
+      _.get(this, 'serverless.service.package.artifact');
+
+    const regex = new RegExp('.*s3.amazonaws.com/(.+)/(.+)');
+    const match = artifactFilePath.match(regex);
+
+    if (match) {
+      return new BbPromise((resolve, reject) => {
+        const tmpDir = this.serverless.utils.getTmpDirPath();
+        const filePath = path.join(tmpDir, match[2]);
+
+        const readStream = S3.getObject({
+          Bucket: match[1],
+          Key: match[2],
+        }).createReadStream();
+
+        const writeStream = fs.createWriteStream(filePath);
+
+        readStream.on('error', (error) => reject(error));
+        readStream.pipe(writeStream)
+          .on('error', (error) => reject(error))
```

Comment:
nit: it could be simplified to `.on('error', reject)`

Reply:
Good suggestion :+1: